### PR TITLE
feature: Add csharp code graph_bfs and graph_dfs

### PR DESCRIPTION
**Pull request by hpstory**
add csharp code graph_bfs and graph_dfs, modify graph adjacency list class inaccessable

If this PR is related to coding or code translation, please read the [contribution guideline](https://github.com/krahets/hello-algo/issues/15#issue-1464069591), fill out the checklist, and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Comment by hpstory** [comment]
bfs 测试结果
bfs_test

dfs 测试结果
dfs_test

File: codes/csharp/chapter_graph/graph_bfs.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+﻿using hello_algo.include;
```
Comment (krahets) [inline codes/csharp/chapter_graph/graph_bfs.cs:1]: Please add the file header and fill in your info.

File: codes/csharp/chapter_graph/graph_dfs.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+﻿using hello_algo.include;
```
Comment (krahets) [inline codes/csharp/chapter_graph/graph_dfs.cs:1]: Same as above

File: codes/csharp/chapter_graph/graph_dfs.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+namespace hello_algo.chapter_graph
+{
+    public class graph_dfs
+    {
+        /* 深度优先遍历 DFS 辅助函数 */
+        public void dfs(GraphAdjList graph, HashSet<Vertex> visited, List<Vertex> res, Vertex vet)
+        {
+            res.Add(vet);     // 记录访问顶点
+            visited.Add(vet); // 标记该顶点已被访问
+                              // 遍历该顶点的所有邻接顶点
```
Comment (krahets) [inline codes/csharp/chapter_graph/graph_dfs.cs:13]: Incorrect indentation

File: codes/csharp/chapter_graph/graph_dfs.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            foreach (Vertex adjVet in graph.adjList[vet])
+            {
+                if (visited.Contains(adjVet))
+                {
+                    continue; // 跳过已被访问过的顶点
+                              // 递归访问邻接顶点
```
Comment (krahets) [inline codes/csharp/chapter_graph/graph_dfs.cs:19]: Incorrect comment position

File: codes/csharp/chapter_graph/graph_bfs.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            {
+                Vertex vet = que.Dequeue(); // 队首顶点出队
+                res.Add(vet);               // 记录访问顶点
+                foreach (Vertex adjVet in graph.adjList[vet])
+                {
+                    if (visited.Contains(adjVet))
+                    {
+                        continue;          // 跳过已被访问过的顶点
+                    }
+
```
Comment (krahets) [inline codes/csharp/chapter_graph/graph_bfs.cs:30]: Remove this empty line

**Review by krahets — COMMENTED** [review]
Thanks for the clean code! Please address the comments.

File: codes/csharp/chapter_graph/graph_bfs.cs (hunk 1 above)
Comment (hpstory) [reply codes/csharp/chapter_graph/graph_bfs.cs]: Fixed

File: codes/csharp/chapter_graph/graph_dfs.cs (hunk 2 above)
Comment (hpstory) [reply codes/csharp/chapter_graph/graph_dfs.cs]: Fixed

File: codes/csharp/chapter_graph/graph_bfs.cs (hunk 5 above)
Comment (hpstory) [reply codes/csharp/chapter_graph/graph_bfs.cs]: Fixed

**Review by krahets — APPROVED** [review]
Thanks!

**Pull request merged**